src/cryptokit.ml

```diff
@@ -2072,6 +2074,98 @@ let new_key ?(rng = Random.secure_rng) ?e numbits =
 
 end
 
+module Paillier = struct
+
+  type key =
+    { size: int;
+      n: string;
+      n2: string;
+      g: string;
+      p: string;
+      q: string;
+      lambda: string;
+      mu: string}
+  let wipe_key k =
+    wipe_string k.n;
+    wipe_string k.n2;
+    wipe_string k.g;
+    wipe_string k.p;
+    wipe_string k.q;
+    wipe_string k.lambda;
+    wipe_string k.mu
+
+  let encrypt key msg ?r ()=
+    let check_random_number r =
+      if (Bn.(relative_prime r (Bn.of_bytes key.n)) && r<(Bn.of_bytes key.n)) then true else false in
+    let rec get_r () =
+      let r = Bn.random ~rng:(Random.secure_rng#random_bytes) (key.size-1) in
+      if check_random_number r then r else get_r () in
+    let r = match r with
+      | None -> get_r ()
+      | Some r -> if check_random_number r then r else raise(Error Wrong_random_number )in
+    let msg = Bn.of_bytes msg in
+    let n = Bn.of_bytes key.n in
+    let n2 = Bn.of_bytes key.n2 in
+    let g = Bn.of_bytes key.g in
+    if Bn.(compare msg zero) <= 0 then raise (Error Negative_msg);
```

Comment:
I'm pretty sure `Bn.of_bytes msg` cannot return negative (< 0) numbers.  It can return 0 if the message is the empty string.  Is that a problem?

Reply:
I didnt know this before. I checked documentation. Yes you are right. There is no issue with 0. I accidentally made it `<=` i guess